### PR TITLE
dev: improving bmath fuzz test

### PR DESCRIPTION
**Pull request by wei3erHase**

### CODE REVIEW

File: test/invariants/fuzz/BMath.t.sol [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    swapFee = clamp(swapFee, MIN_FEE, MAX_FEE);
+
+    uint256 calc_tokenAmountIn =
+      bMath.calcInGivenOut(tokenBalanceOut, tokenWeightOut, tokenBalanceIn, tokenWeightIn, tokenAmountOut, swapFee);
+
+    uint256 calc_tokenAmountOut =
+      bMath.calcOutGivenIn(tokenBalanceOut, tokenWeightOut, tokenBalanceIn, tokenWeightIn, calc_tokenAmountIn, swapFee);
+
+    assert(
+      calc_tokenAmountIn < MIN_BALANCE // return amount too small
```
Comment (0xteddybear) [inline test/invariants/fuzz/BMath.t.sol:86]: I assume the purpose of this check is to skip the assertion if the output is too small which drops us into some edge case where the error is greater, but while testing it I don't seem to find cases where it breaks (perhaps I didn't stumble upon that case *yet* though)

Also I think this discards a whole lot of cases which are valid, I'd remove it if no counter-examples are found, or otherwise would try to use a lower value and try to find out why it breaks

File: test/invariants/fuzz/BMath.t.sol [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    tokenBalanceIn = clamp(tokenBalanceIn, MIN_BALANCE, MAX_BALANCE);
+    swapFee = clamp(swapFee, MIN_FEE, MAX_FEE);
+
+    uint256 calc_tokenAmountIn =
+      bMath.calcInGivenOut(tokenBalanceOut, tokenWeightOut, tokenBalanceIn, tokenWeightIn, tokenAmountOut, swapFee);
+
+    uint256 calc_tokenAmountOut =
+      bMath.calcOutGivenIn(tokenBalanceOut, tokenWeightOut, tokenBalanceIn, tokenWeightIn, calc_tokenAmountIn, swapFee);
+
+    assert(
```
Comment (0xteddybear) [inline test/invariants/fuzz/BMath.t.sol:85]: would it be possible to use foundry's `StdAssertions`?

File: test/invariants/fuzz/BMath.t.sol [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    tokenBalanceIn = clamp(tokenBalanceIn, MIN_BALANCE, MAX_BALANCE);
+    swapFee = clamp(swapFee, MIN_FEE, MAX_FEE);
+
+    uint256 calc_tokenAmountOut =
+      bMath.calcOutGivenIn(tokenBalanceIn, tokenWeightIn, tokenBalanceOut, tokenWeightOut, tokenAmountIn, swapFee);
+
+    uint256 calc_tokenAmountIn =
+      bMath.calcInGivenOut(tokenBalanceIn, tokenWeightIn, tokenBalanceOut, tokenWeightOut, calc_tokenAmountOut, swapFee);
+
+    assert(
```
Comment (0xteddybear) [inline test/invariants/fuzz/BMath.t.sol:54]: I'm not getting any coverage on this assert, all calls seem to revert on the previous line

File: test/invariants/fuzz/BMath.t.sol [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    uint256 tokenBalanceIn,
+    uint256 tokenWeightIn,
+    uint256 tokenBalanceOut,
+    uint256 tokenWeightOut,
+    uint256 tokenAmountIn,
+    uint256 swapFee
+  ) public view {
+    tokenWeightIn = clamp(tokenWeightIn, MIN_WEIGHT, MAX_WEIGHT);
+    tokenWeightOut = clamp(tokenWeightOut, MIN_WEIGHT, MAX_WEIGHT);
+    tokenAmountIn = clamp(tokenAmountIn, MIN_BALANCE, MAX_BALANCE);
```
[the remark this answers is not in the document]
Comment (wei3erHase) [reply test/invariants/fuzz/BMath.t.sol:43]: than the "min token balance"?

[the remark this answers is not in the document]
Comment (0xteddybear) [reply test/invariants/fuzz/BMath.t.sol:43]: :facepalm: got it confused with the one in `BConst`

File: test/invariants/fuzz/BMath.t.sol (hunk 1 above)
Comment (wei3erHase) [reply test/invariants/fuzz/BMath.t.sol]: if we're making a minimum amount to be traded, this assert is making sure that if the output is below that amount we should also discard the case

**Review by 0xteddybear — APPROVED** [review]
